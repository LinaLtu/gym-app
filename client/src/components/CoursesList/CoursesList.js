import React, { useState } from 'react';
import './CoursesList.scss';
import CourseCard from '../CourseCard';

const CSS_NAME = 'course-list'

const mockCourseList = [
    {
        id: 'id1',
        name: 'zumba',
        description: 'latin music and good mood guaranteed',
        startDate: 'monday',
        category: 'c2'
    },
    {
        id: 'id2',
        name: 'body attack',
        description: 'intensive and sweaty',
        startDate: 'tuesday',
        category: 'c3'
    },
    {
        id: 'id3',
        name: 'body balance',
        description: 'low impact',
        startDate: 'friday',
        category: 'c1'
    },
]


const CoursesList = () => {

    const [courseList, setCourseList] = useState(mockCourseList);

    return courseList.map(course => {
        return (
            <div className={CSS_NAME}>
                <CourseCard course={course} />
            </div>
        )

    });
}

export default CoursesList;