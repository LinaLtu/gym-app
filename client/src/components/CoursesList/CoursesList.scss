.course-list {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
    width: 100%;
    padding: 50px;
}