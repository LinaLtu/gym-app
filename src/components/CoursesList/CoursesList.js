import React from 'react';
import './CoursesList.scss';
import CourseCard from '../CourseCard';

const CSS_NAME = 'course-list'

const CoursesList = () => {
    return (
        <div className={CSS_NAME}>
            <CourseCard />
        </div>
    )
}

export default CoursesList;