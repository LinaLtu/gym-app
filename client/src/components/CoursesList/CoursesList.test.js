import React from 'react';
import renderer from 'react-test-renderer';

import CoursesList from '.';

jest.mock('../CourseCard', () => 'CourseCard');

describe('CoursesList Component', () => {
    it('renders according to design', () => {
        const component = renderer.create(
            <CoursesList />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});