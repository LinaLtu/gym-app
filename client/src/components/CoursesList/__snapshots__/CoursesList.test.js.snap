// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard Component renders according to design 1`] = `
<div
  className="course-card"
/>
`;

exports[`CoursesList Component renders according to design 1`] = `<CourseCard />`;
